databricks/resource_databricks_cluster_policy.go

```diff
@@ -0,0 +1,101 @@
+package databricks
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+
+	"github.com/databrickslabs/databricks-terraform/client/model"
+	"github.com/databrickslabs/databricks-terraform/client/service"
+)
+
+func parsePolicyFromData(d *schema.ResourceData) (*model.ClusterPolicy, error) {
+	clusterPolicy := new(model.ClusterPolicy)
+	clusterPolicy.PolicyID = d.Id()
+	if name, ok := d.GetOk("name"); ok {
+		clusterPolicy.Name = name.(string)
+	}
+	if data, ok := d.GetOk("definition"); ok {
+		clusterPolicy.Definition = data.(string)
+	}
+	return clusterPolicy, nil
+}
+
+func resourceClusterPolicyCreate(d *schema.ResourceData, m interface{}) error {
+	client := m.(*service.DBApiClient)
+	clusterPolicy, err := parsePolicyFromData(d)
+	if err != nil {
+		return err
+	}
+	err = client.ClusterPolicies().Create(clusterPolicy)
+	if err != nil {
+		return err
+	}
+	d.SetId(clusterPolicy.PolicyID)
+	return resourceClusterPolicyRead(d, m)
+}
+
+func resourceClusterPolicyRead(d *schema.ResourceData, m interface{}) error {
+	client := m.(*service.DBApiClient)
+	clusterPolicy, err := client.ClusterPolicies().Get(d.Id())
+	if err != nil {
+		return err
+	}
+	err = d.Set("name", clusterPolicy.Name)
+	if err != nil {
+		return err
+	}
+	err = d.Set("definition", clusterPolicy.Definition)
+	if err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
This should handle 404s where a 404 should set Id to zero value

Reply:
https://github.com/databrickslabs/terraform-provider-databricks/pull/129/commits/a076e0d83967ccb95682188579f44244ebefd806 i wonder if this generic way will work for everything